src/Http/AuthenticatesSocialite.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Laravel\Jetstream\Http;
+
+use Illuminate\Contracts\Auth\Authenticatable;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Pipeline;
+use Laravel\Fortify\Actions\AttemptToAuthenticate;
+use Laravel\Fortify\Actions\EnsureLoginIsNotThrottled;
+use Laravel\Fortify\Actions\PrepareAuthenticatedSession;
+use Laravel\Fortify\Actions\RedirectIfTwoFactorAuthenticatable;
+use Laravel\Fortify\Fortify;
+use Laravel\Jetstream\Jetstream;
+use Laravel\Socialite\Contracts\User as SocialiteUserContract;
+
+trait AuthenticatesSocialite
+{
+    /**
+     * Authenticate the user via Socialite.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  string  $provider
+     * @param  \Laravel\Socialite\Contracts\User  $providerUser
+     * @return Pipeline
+     */
+    protected function authenticate(Request $request, string $provider, SocialiteUserContract $providerUser)
+    {
+        // Register the callback to be used for resolving and authenticating the user.
+        Fortify::authenticateUsing(function () use ($provider, $providerUser) {
+            return $this->connectAccount(
+                $this->getUser($providerUser), $provider, $providerUser
+            );
+        });
+
+        return (new Pipeline(app()))->send($request)->through(array_filter([
+            config('fortify.limiters.login') ? null : EnsureLoginIsNotThrottled::class,
+            RedirectIfTwoFactorAuthenticatable::class,
+            AttemptToAuthenticate::class,
+            PrepareAuthenticatedSession::class,
+        ]));
+    }
+
+    /**
+     * Find a user from a given Socialite provided user, or create a new
+     * one from the details given by the provider.
+     *
+     * @param  \Laravel\Socialite\Contracts\User  $providerUser
+     * @return \Illuminate\Contracts\Auth\Authenticatable
+     */
+    protected function getUser(SocialiteUserContract $providerUser)
+    {
```

Comment:
If they connect from settings, I think this would avoid email mismatch issues.

```suggestion
    {
        if(auth()->check()) {
            return auth()->user();
        }
```